webrender/src/tiling.rs

```diff
@@ -680,6 +682,29 @@ impl RenderTask {
 pub const SCREEN_TILE_SIZE: i32 = 256;
 pub const RENDERABLE_CACHE_SIZE: i32 = 2048;
 
+#[derive(Clone, Copy, Debug)]
+pub enum MaskImageSource {
+    User(ImageKey),
+    Renderer(TextureId),
+}
+
+/// Per-batch clipping info merged with the mask image.
+#[derive(Clone, Debug)]
+pub struct MaskedClip {
```

Comment:
I'm still not sure about having this struct. Could we have something like:

```
struct Clip {
  // existing clip rect and corners
  pub mask: Option<MaskImageSource>,
}

And then use that to construct the data to be put in the gpu blocks during prim prepare / cache?

I might be missing something as I haven't actually tried this - but it sounds like it might be a simpler API if it works.